### PR TITLE
Fix Response Timeout (#3216)

### PR DESCRIPTION
**Pull request by Pitometsu**
See the bug report.

### CODE REVIEW

File: plutus-playground-server/src/Playground/Server.hs [hunk 1]
```diff
@@ -111,7 +111,11 @@ initializeServerContext secrets = liftIO $ do
       let localhost = "http://localhost:8009"
       putStrLn $ "WEBGHC_URL not set, using " <> localhost
       parseBaseUrl localhost
-  manager <- newManager defaultManagerSettings
+  manager <- newManager $ defaultManagerSettings
```
Comment (Pitometsu) [inline plutus-playground-server/src/Playground/Server.hs:114]: The actual FIX. The rest is related refactoring.

File: playground-common/src/Control/Monad/Web.hs [hunk 2]
```diff
@@ -22,18 +25,25 @@ class Monad m =>
   doRequest :: Manager -> Request -> m (Either Text (Response LBS.ByteString))
 
 instance MonadWeb IO where
-  makeManager = newManager $ tlsManagerSettings {managerModifyRequest = pure . addUserAgent}
+  makeManager = newManager $ tlsManagerSettings
+    { managerModifyRequest = pure . addUserAgent
+    , managerResponseTimeout = maybe
```
Comment (Pitometsu) [inline playground-common/src/Control/Monad/Web.hs:30]: There's a lot of such duplicated code. However didn't notice that the project actively uses lenses, so decided to keep things simple at the moment.

File: playground-common/src/Control/Monad/Web.hs [hunk 3]
```diff
@@ -22,18 +25,25 @@ class Monad m =>
...
 instance MonadWeb m => MonadWeb (ExceptT e m) where
   makeManager = lift makeManager
   doRequest manager = lift . doRequest manager
 
 instance MonadWeb m => MonadWeb (ReaderT a m) where
   makeManager = lift makeManager
   doRequest manager = lift . doRequest manager
+
+maxInterpretationTime :: Second
+maxInterpretationTime = 160
```
Comment (Pitometsu) [inline playground-common/src/Control/Monad/Web.hs:49]: Probably it's make sense to keep it at the `MonadReader` environment, and read for each app from the shell environment/configuration file/application arguments.

**Review by Pitometsu — COMMENTED** [review]
Please, review and let me know, what's required for merging it.

**Comment by johnalotoski** [comment]
Adding a `hydra-dont-build` label to avoid hydra re-trying to eval this latest commit when there is an error.  Please remove the label when you want hydra to resume trying to build it.

**Comment by ak3n** [comment]
Thanks for the PR!

It changes too many things, so I have picked up only necessary stuff in #3514 and introduced an CLI option.

**Pull request closed without merging**